### PR TITLE
Гусев Никита. Задача 1. Вариант 17. Технология SEQ. Поразрядная сортировка для целых чисел с простым слиянием.

### PR DESCRIPTION
**Pull request by Cufeed**
Этап 1: Разделение на отрицательные и положительные числа
Создать два подмассива:
negatives: хранит абсолютные значения отрицательных чисел
positives: хранит неотрицательные числа (включая ноль)

Этап 2: Сортировка отрицательных чисел
Шаг 1: Отсортировать negatives с помощью Radix Sort (по возрастанию абсолютных значений)
Шаг 2: Развернуть отсортированный массив negatives → порядок убывания абсолютных значений
Шаг 3: Восстановить знак - для всех элементов

Этап 3: Сортировка положительных чисел
Отсортировать positives с помощью Radix Sort (реализация для неотрицательных чисел)

Этап 4: Объединение результатов
Сформировать итоговый массив: отсортированные отрицательные + отсортированные положительные

Особенности:
Отрицательные числа тоже сортируются:
Используется преобразование -num для работы с абсолютными значениями
После сортировки разворот массива обеспечивает правильный порядок (от большего к меньшему по модулю)

Вспомогательные функции:
RadixSortForNonNegative: Сортирует неотрицательные числа поразрядно (LSD)
CountingSort: Стабильная сортировка для отдельного разряда (единицы, десятки и т.д.)
Ключевая операция в CountingSort:
Извлечение цифры: digit = (num / exp) % 10
(Корректно работает только для неотрицательных чисел)

### CODE REVIEW

File: tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "core/task/include/task.hpp"
+#include "seq/gusev_n_sorting_int_simple_merging/include/ops_seq.hpp"
+
```
Comment (petr1core) [inline tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp:10]: Great radix!
Can you add test case for an array, which contains duplicates?

**Review by petr1core — CHANGES_REQUESTED** [review]

File: tasks/seq/gusev_n_sorting_int_simple_merging/data/test.txt
Comment (ermolaevv) [inline tasks/seq/gusev_n_sorting_int_simple_merging/data/test.txt:1]: Why do you need this file?

File: tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      positives.push_back(num);
+    }
+  }
+
+  if (!negatives.empty()) {
+    RadixSortForNonNegative(negatives);
+    std::ranges::reverse(negatives.begin(), negatives.end());
+    for (int& num : negatives) {
+      num = -num;
+    }
```
Comment (ermolaevv) [inline tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp:29]: You can use std::ranges::for_each

File: tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  return true;
+}
+
+bool gusev_n_sorting_int_simple_merging_seq::TestTaskSequential::ValidationImpl() {
+  return task_data->inputs_count[0] == task_data->outputs_count[0];
+}
+
+bool gusev_n_sorting_int_simple_merging_seq::TestTaskSequential::RunImpl() {
+  gusev_n_sorting_int_simple_merging_seq::TestTaskSequential::RadixSort(input_);
+  output_ = input_;
```
Comment (ermolaevv) [inline tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp:89]: Why do you need output_? You can remove it and get rid of unnecessary copying.

File: tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  return true;
+}
+
+bool gusev_n_sorting_int_simple_merging_seq::TestTaskSequential::PostProcessingImpl() {
+  for (size_t i = 0; i < output_.size(); i++) {
+    reinterpret_cast<int*>(task_data->outputs[0])[i] = output_[i];
+  }
```
Comment (ermolaevv) [inline tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp:96]: You can use std::ranges::copy

File: tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  std::vector<int> count(10, 0);
+
+  for (int num : arr) {
+    int digit = (num / exp) % 10;
+    count[digit]++;
+  }
+
+  for (size_t i = 1; i < 10; i++) {
+    count[i] += count[i - 1];
+  }
```
Comment (ermolaevv) [inline tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp:63]: Prefix sum? You can use std::partial_sum

File: tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp
Comment (ermolaevv) [inline tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp:1]: Please add test with random numbers

Comment (ermolaevv) [inline tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp:1]: You can put a lot of duplicated code in functions

File: tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp (hunk 1 above)
Comment (Cufeed) [reply tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp:12]: Thanks. Done

File: tasks/seq/gusev_n_sorting_int_simple_merging/data/test.txt
Comment (Cufeed) [reply tasks/seq/gusev_n_sorting_int_simple_merging/data/test.txt:1]: Deleted

File: tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp (hunk 2 above)
Comment (Cufeed) [reply tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp]: Done

File: tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp (hunk 3 above)
Comment (Cufeed) [reply tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp]: Done

File: tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp (hunk 4 above)
Comment (Cufeed) [reply tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp]: Done

File: tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp (hunk 5 above)
Comment (Cufeed) [reply tasks/seq/gusev_n_sorting_int_simple_merging/src/ops_seq.cpp]: Done

File: tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp
Comment (Cufeed) [reply tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp:1]: Done

**Review by petr1core — APPROVED** [review]

**Review by ermolaevv — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]

File: tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  auto task_data_seq = CreateTaskData(in, out);
+
+  RunT(task_data_seq);
+
+  std::vector<int> expected = in;
+  std::ranges::sort(expected.begin(), expected.end());
+  EXPECT_EQ(expected, out);
+}
+
+TEST(gusev_n_sorting_int_simple_merging_seq, test_radix_sort_random) {
```
Comment (allnes) [inline tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp:80]: please check reverse order

Comment (Cufeed) [reply tasks/seq/gusev_n_sorting_int_simple_merging/func_tests/main.cpp:80]: Done. Added a new test

**Review by allnes — APPROVED** [review]
